include/aikido/planner/vectorfield/MoveEndEffectorTwistVectorField.hpp

```diff
@@ -0,0 +1,92 @@
+#ifndef AIKIDO_PLANNER_VECTORFIELD_MOVEENDEFFECTORTWISTVECTORFIELD_HPP_
+#define AIKIDO_PLANNER_VECTORFIELD_MOVEENDEFFECTORTWISTVECTORFIELD_HPP_
+
+#include "aikido/planner/vectorfield/BodyNodePoseVectorField.hpp"
+#include "aikido/trajectory/Interpolated.hpp"
+
+namespace aikido {
+namespace planner {
+namespace vectorfield {
+
+/// Vector field for moving end-effector by a direction and distance.
+/// It defines a vector field in meta-skeleton state space that moves
+/// an end-effector a desired offset with move-hand-straight constraint.
+/// The move-hand-straight constraint is defined by the direction and
+/// distance range [minimum distance, maximum distance).
+/// Movement less than minimum distance will return failure.
+/// The motion will not move further than maximum distance.
+///
+/// This class defines two callback functions for vectorfield planner.
+/// One for generating joint velocity in MetaSkeleton state space,
+/// and one for determining vectorfield planner status.
+class MoveEndEffectorTwistVectorField : public BodyNodePoseVectorField
+{
+public:
+  /// Constructor
+  ///
+  /// \param[in] stateSpace MetaSkeleton state space.
+  /// \param[in] metaskeleton MetaSkeleton to plan with
+  /// \param[in] bn Body node of end-effector.
+  /// \param[in] direction Unit vector in the direction of motion.
+  /// \param[in] minDistance Minimum distance in meters.
+  /// \param[in] maxDistance Maximum distance in meters. Maximum distance
+  /// should be larger than minimum distance so that a feasible solution
+  /// could be found.
+  /// \param[in] positionTolerance Constraint tolerance in meters.
+  /// \param[in] angularTolerance Constraint tolerance in radians.
+  /// \param[in] maxStepSize The maximum step size used to guarantee
+  /// that the integrator does not step out of joint limits.
+  /// \param[in] jointLimitPadding If less then this distance to joint
+  /// limit, velocity is bounded in that direction to 0.
+  MoveEndEffectorTwistVectorField(
+      aikido::statespace::dart::ConstMetaSkeletonStateSpacePtr stateSpace,
+      ::dart::dynamics::MetaSkeletonPtr metaskeleton,
+      ::dart::dynamics::ConstBodyNodePtr bn,
+      const Eigen::Vector6d& twistSeq,
+      double durationSeq,
+      double positionTolerance,
+      double angularTolerance,
+      double maxStepSize,
+      double jointLimitPadding);
+
+  // Documentation inherited.
+  bool evaluateCartesianVelocity(
+      const Eigen::Isometry3d& pose,
+      Eigen::Vector6d& cartesianVelocity) const override;
+
+  // Documentation inherited.
+  VectorFieldPlannerStatus evaluateCartesianStatus(
+      const Eigen::Isometry3d& pose) const override;
+
+protected:
+  /// Count
+  mutable int mCount{0};
+
+  /// Twist specified
+  Eigen::Vector6d mTwist;
+
+  /// Duration specified
```

Comment:
What does this duration imply? I think a more detailed docstring would be beneficial.